features/issues/components/issue-filter/issue-filter.tsx

```diff
@@ -0,0 +1,161 @@
+import {
+  Button,
+  ButtonSize,
+  ButtonColor,
+  ButtonVariant,
+  ButtonIcon,
+  SelectBox,
+  InputBox,
+  InputIcon,
+} from "@features/ui";
+import { IssueLevel, IssueStatus } from "@api/issues.types";
+import capitalize from "lodash/capitalize";
+import styles from "./issue-filter.module.scss";
+import { useRouter } from "next/router";
+import { useGetIssues } from "@features/issues";
+import { useRef, useEffect, useState } from "react";
+
+export function IssueFilter({ showButton = true }) {
+  const issueLevels = Object.values(IssueLevel).map((level) => ({
+    value: level,
+    label: capitalize(level),
+  }));
+
+  const issueStatus = Object.values(IssueStatus).map((status) => ({
+    value: status,
+    label: capitalize(status),
+  }));
+  const router = useRouter();
+
+  const projectNameFromUrl = Array.isArray(router.query.project)
+    ? router.query.project[0]
+    : router.query.project;
+
+  const [search, setSearch] = useState(projectNameFromUrl || "");
+  const [debouncedSearch, setDebouncedSearch] = useState(
+    projectNameFromUrl || "",
+  );
+
+  const rawStatus = router.query.status;
+  const rawLevel = router.query.level;
+
+  const status =
+    typeof rawStatus === "string" ? (rawStatus as IssueStatus) : undefined;
+  const level =
+    typeof rawLevel === "string" ? (rawLevel as IssueLevel) : undefined;
+
+  const statusRef = useRef<{ setValue: (value: string) => void } | null>(null);
+  const levelRef = useRef<{ setValue: (value: string) => void } | null>(null);
+
+  useEffect(() => {
+    statusRef.current?.setValue(status as string);
+    levelRef.current?.setValue(level as string);
+  }, [status, level]);
```

Comment:
Ah now I see the `setValue` in action. As I said in the select task it would be simpler an more according to best practices to pass the value to the select component as prop. Then you wouldn't need this useEffect as well